### PR TITLE
changes necessary for MuPlus gen and reco

### PR DESCRIPTION
**Pull request by Huangshihua01**
Pull request to merge the changes to produce grid-level simulated Michel e+ events at reduced B-field (50%)

### CODE REVIEW

File: MCDataProducts/inc/GenId.hh [hunk 1]
```diff
@@ -43,7 +43,7 @@ namespace mu2e {
       CaloCalib, InFlightParticleSampler, muplusDecayGun, StoppedMuonXRayGammaRayGun, //37
       cosmicCRY,  pbarFlat, fromAscii, ExternalRMC, InternalRMC, CeLeadingLog, cosmicCORSIKA, //44
       MuCapProtonGenTool, MuCapDeuteronGenTool, DIOGenTool, MuCapNeutronGenTool, // 48
-      MuCapPhotonGenTool, MuCapGammaRayGenTool, CeLeadingLogGenTool,// 50
+      MuCapPhotonGenTool, MuCapGammaRayGenTool, CeLeadingLogGenTool, MuplusMichelGenTool,// 50
```
Comment (kutschke) [inline MCDataProducts/inc/GenId.hh:46]: Please update the comments so that they remain accurate.

Comment (Huangshihua01) [reply MCDataProducts/inc/GenId.hh:46]: I re-write the comment. Now the GenID of 'MuPlusMichelGenTool' is 52 and 'LastEnum' is 53. I committed my changes locally. Should I try pull again?

**Comment by sophiemiddleton** [comment]
@Huangshihua01 I thought we were making the SingleProcessGenerator multi-purpose? Did that not work?

**Comment by sophiemiddleton** [comment]
(I'm not an official reviewer but since I advised Shihua at the start of this): Also, you probably want to remove the TTree from the final version of this. And edit the comment at the top if you really want to duplicate the other module. Perhaps @kutschke can advise whether he thinks duplicating is OK, but I would think making SingleProcessGenerator adaptable for you process would be better.

**Comment by kutschke** [comment]
@FNALbuild run build test

**Comment by kutschke** [comment]
> As Sophie and I discussed I will make some further changes and make it meet the standard to be pushed. At this time I will close this request.

Thanks for letting us know.  Remember that you can close it now, keep pushing to the same branch, and reopen it when ready.

**Comment by kutschke** [comment]
Per the comments above, I am closing this PR.  It can be reopened when ready.

**Pull request closed without merging**